src/pages/api/progress.ts

```diff
@@ -0,0 +1,85 @@
+import TinCan from 'tincanjs'
+import axios from 'axios'
+import {NextApiRequest, NextApiResponse} from 'next'
+import {connectScrollTo} from 'react-instantsearch-dom'
+
+const SCORM_CLOUD_ENDPOINT = `https://cloud.scorm.com/lrs/PZ9CYEKKV8/`
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  console.log(req.query)
+  if (req.method === 'GET') {
+    try {
+      const lrs = new TinCan.LRS({
+        endpoint: SCORM_CLOUD_ENDPOINT,
+        username: process.env.SCORM_CLOUD_USERNAME,
+        password: process.env.SCORM_CLOUD_PASSWORD,
+        allowFail: false,
+      })
+
+      const verb = req.query.verb
+
+      const statement: any = new TinCan.Statement({
```

Comment:
https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md

They have a very well defined spec for creating these activity statements. They are considered immutable, similar to Redux I guess or any event sourcing type of system.

The statements are evaluated against a schema, but if you follow the rules they are super flexible.

I like their very robust and well defined set of verbs and think their general advice not to make up new ones makes a lot of sense 👍 

http://xapi.vocab.pub/verbs/index.html

https://xapi.com/blog/deep-dive-verb/